test/jdk/java/nio/channels/DatagramChannel/SelectWhenRefused.java

```diff
@@ -49,43 +57,69 @@ public static void main(String[] args) throws IOException {
             dc.configureBlocking(false);
             dc.register(sel, SelectionKey.OP_READ);
 
-            /* Test 1: not connected so ICMP port unreachable should not be received */
-            sendDatagram(dc, refuser);
-            int n = sel.select(2000);
-            if (n > 0) {
-                sel.selectedKeys().clear();
-                // BindException will be thrown if another service is using
-                // our expected refuser port, cannot run just exit.
-                DatagramChannel.open().bind(refuser).close();
-                throw new RuntimeException("Unexpected wakeup");
+            for (int i = 0; i < MAX_TRIES; i++) {
+                /* Test 1: not connected so ICMP port unreachable should not be received */
+                sendDatagram(dc, refuser);
+                int n = sel.select(2000);
+                if (n > 0) {
+                    ignoreStrayWakeup = checkUnexpectedWakeup(dc, sel.selectedKeys());
+                    sel.selectedKeys().clear();
+                    if (ignoreStrayWakeup) {
+                        if (i < MAX_TRIES - 1) continue;
+                    }
+                    // BindException will be thrown if another service is using
+                    // our expected refuser port, cannot run just exit.
+                    DatagramChannel.open().bind(refuser).close();
+                    throw new RuntimeException("Unexpected wakeup");
+                } else {
+                    break;
+                }
```

Comment:
nit: this is OK - but in fact the else isn't even needed. My bad. We could just break after the if, whether we entered it or not, because if we reach here it means we have found what we expected: all other paths result in continue or throw. So this could just be:
```
if (n > 0) { ...
}
break;
```